pkg/ingester/ingester.go

```diff
@@ -2079,10 +2080,35 @@ func wrappedTSDBIngestErr(ingestErr error, timestamp model.Time, labels []mimirp
 		return nil
 	}
 
-	return fmt.Errorf(errTSDBIngest, ingestErr, timestamp.Time().UTC().Format(time.RFC3339Nano), mimirpb.FromLabelAdaptersToLabels(labels).String())
+	var errID globalerror.ID
+	var errMsg string
+	switch cause := errors.Cause(ingestErr); cause {
+	case storage.ErrOutOfBounds:
+		errID = globalerror.OutOfBounds
+		errMsg = "something out of bounds"
```

Comment:
I'm not a big fan of having to remap it here. Given we already have code above handling these cases (see around line `675`), can't we just return custom errors there?